### PR TITLE
feat: Set ownerReference for vulnerabilities resource

### PR DESCRIPTION
**Pull request by danielpacak**
Resolves: #26

Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: pkg/kube/object.go [hunk 1]
```diff
@@ -84,3 +88,113 @@ func (ci ContainerImages) AsJSON() (string, error) {
...
+	return nil
+}
+
+func upsertOwnerRef(ref metav1.OwnerReference, object metav1.Object) {
+	owners := object.GetOwnerReferences()
+	idx := indexOwnerRef(owners, ref)
+	if idx == -1 {
+		owners = append(owners, ref)
+	} else {
+		owners[idx] = ref
```
Comment (knqyf263) [inline pkg/kube/object.go:172]: Just for my understanding:
In this case, what field will be updated? Only `UID` considering the unit test? I'm curious when it happens. It looks like `Group`, `Kind`, and `Name` are always the same.

**Review by knqyf263 — APPROVED** [review]

File: pkg/kube/object.go (hunk 1 above)
Comment (danielpacak) [reply pkg/kube/object.go:172]: In this case all properties of the given `metav1.OwnerReference` are updated including GVK even though it's very unlikely they change in real Starboard scenario. However this utility method is generic to cater for cases where we have some other controller / Kubernetes app that also relies on the `ownerReferences` property. In this case the GVK might be different.

Comment (knqyf263) [reply pkg/kube/object.go:172]: If `indexOwnerRef` returns `idx` other than `-1`, it means GVK is the same, right? So, I wanted to know why `owners[idx] = ref` is necessary. If all properties are the same as existing `ownerReference`, we don't need to update it. But `UID` might be different when a user deletes the resource and recreates it with the same GVK and name.

Comment (danielpacak) [reply pkg/kube/object.go:172]: Yup. You're right. It might only happen if the UID has changed.

Comment (knqyf263) [reply pkg/kube/object.go:172]: I see. Thank you for the clarification!

**Pull request merged**